lib/open_food_network/orders_and_fulfillments_report.rb

```diff
@@ -238,31 +282,55 @@ def columns
           proc { |line_items| line_items.first.order.bill_address.andand.state }
         ]
       else
-        [proc { |line_items| line_items.first.product.supplier.name },
-         proc { |line_items| line_items.first.product.name },
-         proc { |line_items| line_items.first.full_name },
-         proc { |line_items| line_items.sum(&:quantity) },
-         proc { |line_items| line_items.first.price },
-         proc { |line_items| line_items.sum { |li| li.quantity * li.price } },
-         proc { |_line_items| "" },
-         proc { |_line_items| I18n.t(:report_header_incoming_transport) }]
+        DefaultReport.new(self).columns
       end
     end
 
     private
 
+    def supplier_name
+      proc { |line_items| find_variant(line_items.first.variant_id).product.supplier.name }
+    end
+
+    def product_name
+      proc { |line_items| find_variant(line_items.first.variant_id).product.name }
+    end
+
+    def line_item_name
+      proc { |line_item| find_variant(line_item.variant_id).full_name }
+    end
+
+    def line_items_name
+      proc { |line_items| find_variant(line_items.first.variant_id).full_name }
+    end
+
```

Comment:
I aimed at DRYing the tones of duplication (which led me to too many stupid and missed cases that needed fixing) but it required too much time. Until here, we improved a bit.